setup.py

```diff
@@ -338,7 +339,13 @@ def run(self):
     extra_compile_args += ['-DWITH_NCCL']
 
 if WITH_CUDNN:
-    main_libraries += ['cudnn']
+    CUDNN_LIB_NAME = 'libcudnn.so'
+    if platform.system() == 'Darwin':
+        CUDNN_LIB_NAME = 'libcudnn.dylib'
+    CUDNN_LIB = os.path.join(CUDNN_LIB_DIR, CUDNN_LIB_NAME)
+    CUDNN_LIB = os.path.realpath(CUDNN_LIB)
+
+    main_link_args += [ CUDNN_LIB ]
```

Comment:
ldd looks fine:
```
libcudnn.so.5 => /usr/local/cuda/lib64/libcudnn.so.5 (0x00007f5d7e297000)
```

However, it's weird that it links against `libcudnn.so.5`, because the setup reports linking to:

```
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.5
```